bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/types/QuantityTypeTest.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.library.types;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import tec.uom.se.quantity.QuantityDimension;
+import tec.uom.se.unit.Units;
+
+/**
+ * @author Gaël L'hopital
+ */
+
+public class QuantityTypeTest {
+
+    @Test
+    public void testDimensionless() {
+        QuantityType dt0 = new QuantityType("12");
+        assertTrue(dt0.getUnit().getDimension() == QuantityDimension.NONE);
+    }
+
+    @Test
+    public void testKnownInvalidConstructors() {
+        // These fails but shall not in principle, they are specific
+        // cases of dimensionless values
+
+        try {
+            @SuppressWarnings("unused")
+            QuantityType dt5 = new QuantityType("2°");
```

Comment:
You may only call the c'tor here w/o assignment. Then you dont have to suppress the warning.